src/commands/setup.ts

```diff
@@ -0,0 +1,235 @@
+}
+
+export const PROJECT_NAME = "BedrockRocks1";
+
+/**
+ * Prompts for questions
+ *
+ * @return answers to the questions
+ */
+export const prompt = async (): Promise<IAnswer> => {
+  const questions = [
+    {
+      message: "Enter organization name\n",
+      name: "azdo_org_name",
+      type: "input",
+      validate: validateOrgName
+    },
+    {
+      default: PROJECT_NAME,
+      message: "Enter name of project to be created\n",
+      name: "azdo_project_name",
+      type: "input",
+      validate: validateProjectName
+    },
+    {
+      mask: "*",
+      message: "Enter your Azure DevOps personal access token\n",
+      name: "azdo_pat",
+      type: "password",
+      validate: validateAccessToken
+    }
+  ];
+  const answers = await inquirer.prompt(questions);
+  return {
+    azdo_org_name: answers.azdo_org_name as string,
+    azdo_pat: answers.azdo_pat as string,
+    azdo_project_name: answers.azdo_project_name as string
+  };
+};
+
+export const getAnswerFromFile = (file: string): IAnswer => {
+  let content = "";
+  try {
+    content = fs.readFileSync(file, "utf-8");
+  } catch (_) {
+    throw new Error(
+      `${file} did not exist or not accessible. Make sure that it is accessible.`
+    );
+  }
+
+  const arr = content.split("\n").filter(s => s.trim().length > 0);
+  const map: { [key: string]: string } = {};
+  arr.forEach(s => {
+    const idx = s.indexOf("=");
+    if (idx !== -1) {
+      map[s.substring(0, idx).trim()] = s.substring(idx + 1).trim();
+    }
+  });
+  map.azdo_project_name = map.azdo_project_name || PROJECT_NAME;
+
+  const vOrgName = validateOrgName(map.azdo_org_name);
+  if (typeof vOrgName === "string") {
+    throw new Error(vOrgName);
+  }
+
+  const vProjectName = validateProjectName(map.azdo_project_name);
+  if (typeof vProjectName === "string") {
+    throw new Error(vProjectName);
+  }
+
+  const vToken = validateAccessToken(map.azdo_pat);
+  if (typeof vToken === "string") {
+    throw new Error(vToken);
+  }
+
+  return {
+    azdo_org_name: map.azdo_org_name,
+    azdo_pat: map.azdo_pat,
+    azdo_project_name: map.azdo_project_name
+  };
+};
+
+/**
+ * Creates SPK config file under `user-home/.spk` folder
+ *
+ * @param answers Answers provided to the commander
+ */
+export const createSPKConfig = (answers: IAnswer) => {
+  const data = yaml.safeDump({
+    azure_devops: {
+      access_token: answers.azdo_pat,
+      org: answers.azdo_org_name,
+      project: answers.azdo_project_name
+    }
+  });
+  fs.writeFileSync(defaultConfigFile(), data);
+};
+
+/**
+ * Creates an Azure DevOps Project.
+ *
+ * @param coreAPI Core API service
+ * @param name Name of Project
+ */
+export const createProject = async (coreAPI: ICoreApi, name: string) => {
+  try {
+    await coreAPI.queueCreateProject({
+      capabilities: {
+        processTemplate: {
+          templateTypeId: "27450541-8e31-4150-9947-dc59f998fc01" // TOFIX: do not know what this GUID is about
```

Comment:
Why is this guid needed?

Reply:
this is something that I cannot figure out its value; and I googled for it and found that this value is used. I asked Andre to found a contact where we can ask.

Reply:
Got it